shoebot/gui/gtk_window.py

```diff
@@ -265,13 +265,20 @@ def var_changed(self, name, value):
 
     def output_image_filename(self, format):
         """
-        :param format:  Format, e.g. svg, pdf, png
+        :param format:  Format: "svg", "pdf", "png" or "ps"
         :return:  Full image filename, based on bot name
         """
         script = self.bot._namespace["__file__"]
-        frame = self.bot._namespace["ITERATION"]
+
         if script:
-            return f"{Path(script).stem}-{frame:04}.{format}"
+            filepath = Path(f"{Path(script).stem}.{format}")
+            if filepath.exists():
+                filecounter = 1
```

Comment:
We could cache the counters, here's a hacky way:

`filecounter = getattr(self, f"filecounter_{format}", 1)`

Then at the end:

`setattr(self, f"filecounter_{format}", filecounter)`

This could be de-hackified by refactoring, I'm not going to get time this weekend though - and this is probably nicer than this taking longer and longer each time we output an image.

Reply:
hm i'm not sure i get how this works, i'll wait for you to be back and i'll pester you on matrix :)

Reply:
It's getting a value from a variable called something like `self.filecounter_png` or `self.filecounter_svg`.

If the variable doesn't exist, it's defaulting to `1`.

Then the setattr line saves it.